tests/smoke-tests/src/test/java/org/apache/activemq/artemis/tests/smoke/checkTest/CheckTest.java

```diff
@@ -0,0 +1,350 @@
+
+      QueueCheck queueCheck;
+
+      try {
+         queueCheck = new QueueCheck();
+         queueCheck.setUser("admin");
+         queueCheck.setPassword("admin");
+         queueCheck.setName(queueName);
+         queueCheck.execute(ACTION_CONTEXT);
+
+         Assert.fail("CLIException expected.");
+      } catch (Exception e) {
+         Assert.assertTrue("CLIException expected.", e instanceof CLIException);
+      }
+
+      ServerLocator locator = ServerLocatorImpl.newLocator("tcp://localhost:61616");
+      try (ClientSessionFactory factory = locator.createSessionFactory(); ClientSession session = factory.createSession()) {
+         session.createQueue(new QueueConfiguration(queueName).setRoutingType(RoutingType.ANYCAST));
+      }
+
+      try {
+         queueCheck = new QueueCheck();
+         queueCheck.setUser("admin");
+         queueCheck.setPassword("admin");
+         queueCheck.setName(queueName);
+         queueCheck.setConsume(1);
+         queueCheck.setTimeout(100);
+         queueCheck.execute(ACTION_CONTEXT);
+
+         Assert.fail("CLIException expected.");
+      } catch (Exception e) {
+         Assert.assertTrue("CLIException expected.", e instanceof CLIException);
+      }
+
+
+      queueCheck = new QueueCheck();
+      queueCheck.setUser("admin");
+      queueCheck.setPassword("admin");
+      queueCheck.setName(queueName);
+      Assert.assertEquals(1, queueCheck.execute(ACTION_CONTEXT));
+
+      queueCheck = new QueueCheck();
+      queueCheck.setUser("admin");
+      queueCheck.setPassword("admin");
+      queueCheck.setUp(true);
+      queueCheck.setName(queueName);
+      Assert.assertEquals(1, queueCheck.execute(ACTION_CONTEXT));
+
+      ConnectionFactory cf = CFUtil.createConnectionFactory("core", "tcp://localhost:61616");
+
+      final int messages = 3;
+
+      queueCheck = new QueueCheck();
+      queueCheck.setUser("admin");
+      queueCheck.setPassword("admin");
+      queueCheck.setName(queueName);
+      queueCheck.setBrowse(null);
+      Assert.assertEquals(1, queueCheck.execute(ACTION_CONTEXT));
+
+
+      queueCheck = new QueueCheck();
+      queueCheck.setUser("admin");
+      queueCheck.setPassword("admin");
+      queueCheck.setName(queueName);
+      queueCheck.setConsume(null);
+      Assert.assertEquals(1, queueCheck.execute(ACTION_CONTEXT));
+
+
+      try (Connection connection = cf.createConnection(); Session session = connection.createSession(true, Session.SESSION_TRANSACTED)) {
+         MessageProducer producer = session.createProducer(session.createQueue(queueName));
+         for (int i = 0; i < messages; i++) {
+            System.out.println("Sending " + i);
+            TextMessage message = session.createTextMessage("hello " + i);
+            message.setStringProperty("local", String.valueOf(i));
+            producer.send(message);
+         }
+         session.commit();
+         Wait.assertEquals(messages, () -> getMessageCount(simpleManagement, queueName));
```

Comment:
Using the default 30 seconds of wait with 100ms periods seems excessive here....and everywhere below.

Reply:
It's never happening though. I like the large timeout for debugging. 

I've tried to change the default in the past.. but I don't think we need to individualize things here.

Reply:
Its never happening until it does. Thats the point of adding a timeout. Checks should have action-appropriate timeouts so things fail in a timely manner when issues do arise and not taking e.g 30sec for no good reason. Even one 100ms delay if it isnt ready on the very first check is likely wasting time really.

(E.g, remember that AIO check thing? Until I added the fail-fast bailou , whenever that issue arose it used to take hours for the test job to actually fail, as it sat waiting unnecessarily for a full 30sec on every check).

Reply:
These are all simple checks with message counts.. I don't think we need to do much more about this.

Reply:
@gemmellr the long default on Wait is a general concern. we should rather change that default. I don't think we should make it different here.

I have tried before but it was a large task that I postponed. I would merge this as is and tackle the large timeout eventually. 


I think the default timeout should be 1 second (5 seconds, the very largest). The main idea of Wait was async acts and sends on queue.. so a second should be enough.. anything other than that should have a separate method call. We should do the opposite here.

Reply:
Precisely, they are just simple checks, expected to pass immediately or almost...so giving them 30 seconds does not seem like an action-appropriate timeout.

Similarly, if somehow they didnt pass on the very first check, burning 100ms at a time until they do is potentially a significant waste of time sat doing nothing. The overall test suite is damn slow, we really should be paying attention to optimising such things.

Reply:
I would personally remove the default wait entirely and have checks set an appropriate timeout and period for whatever they are doing.

Reply:
This is not part of this change though... These sleeps were already part of the previous test.

I will add a timeout, but it shouldn't be subject of this task now.

Reply:
The default timeout makes it very simple to convert an Assert.assertEquals into an Wait.assertEquals.

However the 30seconds wait timeout is a mistake that we should fix.


that's a separate change.. I added the timeout now.

Reply:
> This is not part of this change though... These sleeps were already part of the previous test.

It seems like you added the waits here, in which case yes its subject to this task.

> The default timeout makes it very simple to convert an Assert.assertEquals into an Wait.assertEquals.

Maybe, but people shouldnt really be adding timeouts without looking at what the action-appropriate timeout and retry interval actually is.